core/services/ocrcommon/transmitter.go

```diff
@@ -62,11 +77,11 @@ func (t *transmitter) FromAddressForTransaction() common.Address {
 	// Use Round-Robin to select the next fromAddress.
 	nextFromAddress := t.fromAddresses[t.nextFromAddressIndex]
 
-	// Only apply round-robin logic for multiple sending keys.
-	if len(t.fromAddresses) > 1 {
-		t.nextFromAddressIndex++
-		if t.nextFromAddressIndex >= len(t.fromAddresses) {
-			t.nextFromAddressIndex = 0
+	// Only apply round-robin logic for multiple sending keys and a valid keystore.
+	if len(t.fromAddresses) > 1 && t.ethKeyStore != nil {
```

Comment:
We should fail here if fromaddress>1 and keystore is nil

Reply:
Probably fail on NewTransmitter call so we fail job creation

Reply:
This makes sense but we need to make sure that it doesn't break existing jobs when nops upgrade (seems like this is used in OCR1 as well)